src/libraries/Common/src/Interop/Unix/System.Native/Interop.GetGroups.cs

```diff
@@ -0,0 +1,49 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class Sys
+    {
+        internal static unsafe uint[]? GetGroups()
+        {
+            const int InitialGroupsLength =
+#if DEBUG
+                1;
+#else
+                64;
+#endif
+            Span<uint> groups = stackalloc uint[InitialGroupsLength];
```

Comment:
I assume the chances someone actually has more than 64 groups is exceedingly small?

Reply:
Yes, 64 is plenty.